src/components/context/TodosContext.tsx

```diff
@@ -0,0 +1,207 @@
+import React, {
+  Dispatch,
+  SetStateAction,
+  useContext,
+  useEffect,
+  useMemo,
+  useState,
+} from 'react';
+
+import * as todosServices from '../../api/api';
+import { useError } from './ErrorContext';
+import { Status, TodoError } from '../../types/enums';
+import { Todo } from '../../types/Todo';
+import { USER_ID } from '../../utils/todos';
+
+interface TodosContextType {
+  todos: Todo[];
+  statusTodo: Status;
+  tempTodo: Todo | null;
+  isProcessing: number[];
+  setIsProcessing: Dispatch<SetStateAction<number[]>>;
+  setTempTodo: (_todo: Todo) => void;
+  removeTodo: (_todoId: number) => Promise<void>;
+  updateTodo: (_updatedTodo: Todo) => Promise<void>;
+  addTodo: (_todo: Todo) => Promise<void>;
+  toggleOne: (_updatedTodo: Todo) => void;
+  handleClearCompleted: () => void;
+  toggleAll: () => void;
+  setStatusTodo: (_statusTodo: Status) => void;
+}
+
+const contextValue: TodosContextType = {
+  todos: [],
+  statusTodo: Status.All,
+  tempTodo: null,
+  isProcessing: [],
+  setIsProcessing: () => {},
+  setTempTodo: () => {},
+  removeTodo: async () => {},
+  updateTodo: async () => {},
+  addTodo: async () => {},
+  toggleOne: async () => {},
+  handleClearCompleted: async () => {},
+  toggleAll: () => {},
+  setStatusTodo: () => {},
+};
+
+export const TodosContext = React.createContext<TodosContextType>(contextValue);
+
+interface Props {
+  children: React.ReactNode;
+}
+
+export const TodosProvider: React.FC<Props> = ({ children }) => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [statusTodo, setStatusTodo] = useState(Status.All);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [isProcessing, setIsProcessing] = useState<number[]>([]);
```

Comment:
It's not a boolean variable, `loadingTodoIds` describes better or something similar